stm32-modules/heater-shaker/firmware/heater_task/freertos_heater_task.cpp

```diff
@@ -35,21 +42,75 @@ static FreeRTOSMessageQueue<heater_task::Message>
 // NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
 static auto _task = heater_task::HeaterTask(_heater_queue);
 
+// NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
+static constexpr uint32_t _hardware_stack_size = 128;
+// NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
+static std::array<StackType_t, _hardware_stack_size> _hardware_stack;
+// NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
+static StaticTask_t _hardware_data;
+
+struct HeaterTasks {
+    // NOLINTNEXTLINE(misc-non-private-member-variables-in-classes)
+    heater_hardware hardware;
+    // NOLINTNEXTLINE(misc-non-private-member-variables-in-classes)
+    heater_task::HeaterTask<FreeRTOSMessageQueue> heater_main_task;
+    // NOLINTNEXTLINE(misc-non-private-member-variables-in-classes)
+    TaskHandle_t hardware_task_handle;
+    HeaterTasks(FreeRTOSMessageQueue<heater_task::Message> &queue,
+                void (*conversion_handler)(conversion_results *))
+        : hardware{.conversions_complete = conversion_handler,
+                   .hardware_internal = nullptr},
+          heater_main_task(queue),
+          hardware_task_handle(nullptr) {}
+};
+
+// NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
+HeaterTasks _heater_tasks(_heater_queue, handle_conversion);
+
+static void handle_conversion(conversion_results *results) {
```

Comment:
Should the input be `const conversion_results * results`? Does it matter?

Reply:
Yeah it definitely should be, good catch. It doesn't  really matter that much but it's good practice.